swift/LottieRenderer.swift

```diff
@@ -50,15 +51,49 @@ public class LottieRenderer {
         lottie.animation.setFrame(frameIndex)
 
         let picture = lottie.animation.getPicture()
-        picture.resize(canvas.size)
-        picture.resetTransform()
-        picture.stretchToFit(contentRect)
+        let size = picture.getSize()
 
-        let radians = rotation * .pi / 180.0
-        picture.apply(transform: CGAffineTransform(rotationAngle: radians))
+        // Create the transform for the Picture by multiplying three transform matrices.
+        // 1. Translate the Picture to the content rect origin.
+        // 2. Scale the Picture based on the Picture's size versus the content rect size.
+        // 3. Rotate the Picture about its center.
+        let transform =
+            CGAffineTransform(
+                translationX: -contentRect.minX,
+                y: -contentRect.minY
+            )
+            .concatenating(
+                CGAffineTransform(
+                    scaleX: size.width / contentRect.width,
+                    y: size.height / contentRect.height
+                )
+            )
+            .concatenating(
+                CGAffineTransform(
+                    rotationAngle: rotation * .pi / 180.0
+                )
+                .about(pivotPoint:
+                    CGPoint(
+                        x: size.width / 2,
+                        y: size.height / 2
+                    )
+                )
+            )
+        
+        picture.setTransform(transform)
```

Comment:
@rlaguilar thanks for the great comment around the transform logic. I agree - it was getting quite messy. So I have taken steps to improve it. 

What I've done is make the transform logic much clearer, creating one single transform matrix that is applied to the `Picture` once. No longer need to deal with getting and setting the matrix multiple times.

And here I've just used pure `CGAffineTransforms`, and am multiplying them in the specified order we need. I was a bit unsure to about putting most of the logic in this function, but I think it actually reads quite nicely. 

Let me know what you think!